### PR TITLE
RavenDB-17281 Query explanations in studio are ordered by id, even if…

### PR DESCRIPTION
**Pull request by ml054**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17281

### Additional description

Retain server side order for explanations 

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### CODE REVIEW

File: src/Raven.Studio/typescript/viewmodels/database/query/query.ts [hunk 1]
```diff
@@ -1197,14 +1197,21 @@ class query extends viewModelBase {
     }
     
     private onExplanationsLoaded(explanations: dictionary<Array<string>>) {
-        _.forIn(explanations, (doc, id) => {
-            this.explanationsCache.set(id, {
-               id: id,
-                explanations: doc
-            });
-        });
+        for (const id of Object.keys(explanations)) {
```
Comment (ayende) [inline src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: I don't think that we ensure the order of items in a dictionary.  Can we do this loop by reading the value from the `results`, which are ensured to be sorted?

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: this dict comes from server - and it is json object, where order is ensured acording to ECMAscript spec. So if server returns proper order then we have correct order.

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: ok looks like we use regular dict on server side, 
https://github.com/ravendb/ravendb/blob/e79e7c5ec32dc753a3f30fb3216887be7a3ae26e/src/Raven.Server/Documents/Queries/DocumentQueryResult.cs#L102

so I need to iterate over results to obtain order

Thanks @ayende

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/query/query.ts]: Updated logic - I hope it makes sense right now. 

I'm sorting using results if possible.

**Pull request merged**